spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/source/SparkBatchQueryScan.java

```diff
@@ -162,13 +162,14 @@ public void filter(Predicate[] predicates) {
     }
   }
 
-  protected Map<String, DeleteFileSet> rewritableDeletes() {
+  protected Map<String, DeleteFileSet> rewritableDeletes(boolean forDVs) {
     Map<String, DeleteFileSet> rewritableDeletes = Maps.newHashMap();
 
     for (ScanTask task : tasks()) {
       FileScanTask fileScanTask = task.asFileScanTask();
       for (DeleteFile deleteFile : fileScanTask.deletes()) {
-        if (ContentFileUtil.isFileScoped(deleteFile)) {
+        // Both file-scoped and partition-scoped position deletes must be rewritten for DVs
+        if (forDVs || ContentFileUtil.isFileScoped(deleteFile)) {
```

Comment:
It would be nice to also cover this with tests.

Reply:
Ah I fixed to avoid broadcasting eq. deletes unnecessarily but need to figure out how to add a test which would inspect the broadcast contents, will a publish a separate PR for that